content/doc/book/pipeline/syntax.adoc

```diff
@@ -485,6 +479,22 @@ automatically be defined: `MYVARNAME_USR` and `MYVARNAME_PSW` respectively.
 | Inside the `pipeline` block, or within `stage` directives.
 |===
 
+===== Supported Credentials Type
+
+Secret Text:: 
+the environment variable specified will be set to the Secret Text content
+Secret File::
+the environment variable specified will be set to the location of the File
+file that is temporarily created
+Username and password:: 
+the environment variable specified will be set to `username:password` and two
+additional environment variables will be automatically defined: `MYVARNAME_USR`
```

Comment:
I know this text is moved from the previous block, but I don't fully understand what is being described.

Is the variable name always `MYVARNAME_PSW` for the password?  I didn't find any references to that string in my search of the Jenkins plugins, so I'm suspicious that some argument is used to define the variables which will receive the value of the username and password.

Reply:
> Is the variable name always MYVARNAME_PSW for the password?

It is the name of the variable you define in the directive, with `_PSW` appended.

The directive works like the following:

```
environment {
    MYVARNAME = credentials('my-prefined-secret')
}
```

Depending of the type of credentials, this results in many variables being injected. 
For Username / Password, you have three `MYVARNAME` (username and password in the form `username:password`), `MYVARNAME_USR` and `MYVARNAME_PSW`. 
For SSH with Private Key you have three also `MYVARNAME` (temp location of the private key file), `MYVARNAME_USR` and `MYVARNAME_PSW` (passphrase).

Reply:
Could that directive be included in the documentation?

I would not have inferred that the return type from the credentials step is multi-valued and the number of values, names of the values, and content of the values depends on the type of the credential.

Reply:
Is it worth describing what will happen in case of an unsupported credential type?